l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py

```diff
@@ -1124,11 +1126,28 @@ def compute_xml_amount_untaxed(self, cr, uid, DatiRiepilogo, context=None):
             amount_untaxed += float(Riepilogo.ImponibileImporto)
         return amount_untaxed
 
+    def compute_xml_amount_total(self, cr, uid, DatiRiepilogo, context=None):
+        amount_total = 0.0
+        for Riepilogo in DatiRiepilogo:
+            amount_total += float(Riepilogo.ImponibileImporto)
+            amount_total += float(Riepilogo.Imposta)
+            # 2.2.2.3
+            if Riepilogo.SpeseAccessorie:
+                amount_total += float(Riepilogo.SpeseAccessorie)
+            # 2.2.2.4 TODO arrotondamento
+        return amount_total
+
     def check_invoice_amount(
         self, cr, uid, invoice, FatturaElettronicaBody, context=None
     ):
         if context is None:
             context = {}
+        xml_total = self.compute_xml_amount_total(
+            cr, uid,
+            FatturaElettronicaBody.DatiBeniServizi.DatiRiepilogo,
+            context=context)
+
+        invoice.write({'check_total': xml_total}, context=context)
```

Comment:
Eh si di fatto perche' se guardi compute_xml_amount_total, aggiunge anche le spese accessorie 2.2.2.3, che possono fare cambiare il totale della fattura.
Rimane comunque da capire come comprendere in quel metodo il campo 2.2.2.4 arrotondamento, e se il valore che puo' contenere e' da sommare o sottrarre alla totale.